src/vsg/viewer/Viewer.cpp

```diff
@@ -211,18 +212,21 @@ void Viewer::handleEvents()
 class CollectSecondaryCommandGraph : public ConstVisitor
 {
 public:
-    vsg::CommandGraphs _secondaries;
+    vsg::CommandGraphs secondaries;
+    std::vector<std::shared_ptr<std::mutex> > execCommandMuters;
```

Comment:
Are you sure you want to use the name Muters?

The dictionary definition of Muter is:

"adjective, mut·er, mut·est.
silent; refraining from speech or utterance.
not emitting or having sound of any kind."

If you are looking for plural of mutex then mutices or mutexes, or move the plural elsewhere listOfExceCommandMutex.

Reply:
as a mutex mutes the thread, it's not absurd to call it a muter...However if you want me to change this, i don't mind

Reply:
done

Reply:
Halting a thread isn't muting it.  Muting would be something like redirect console output to /dev/null, or muting the sound of video while the video continues playing.

If you have a list of mutex then just call it that, or call it by a proper plural.

In terms of design it might be that sharing a std::mutex is not the ideal thing anyway so this discussion might a "mute" point :-)

However, as I haven't done a long enough review of your changes and thought about the wider topic of viewer threading I can't provide guidance yet on whether the direction you are going is ideal.

I have other work to complete this week before I can dedicate any time to the topic of getting subpass integrated and to look at viewer threading.